testing/tenant.go

```diff
@@ -599,49 +645,46 @@ func Delete(t *testing.T, init func(*testing.T, TenantFields) (influxdb.TenantSe
 			t.Errorf("expected 1 buckets, got: %v", bs)
 		}
 		// Now delete user1 -> bucket2.
-		// bucket2 should be deleted (nobody points to it).
+		// Still expect bucket2 to exist (nobody points to it).
 		if err := s.DeleteUserResourceMapping(ctx, data.Buckets[1].ID, data.Users[0].ID); err != nil {
 			t.Fatal(err)
 		}
 		bs, nbs, err = s.FindBuckets(ctx, influxdb.BucketFilter{ID: &data.Buckets[1].ID})
 		if err != nil {
 			t.Fatal(err)
 		}
-		if nbs > 0 {
-			t.Errorf("expected no buckets, got: %v", bs)
-			urms, _, err := s.FindUserResourceMappings(ctx, influxdb.UserResourceMappingFilter{
-				ResourceType: influxdb.BucketsResourceType,
-				ResourceID:   data.Buckets[1].ID,
-			})
-			if err != nil {
-				t.Fatal(err)
-			}
-			t.Logf("bucket2 should be dangling at this point. These are its urms: %+v", urms)
+		if nbs != 1 {
+			t.Errorf("expected 1 buckets, got: %v", bs)
+		}
+		urms, nurms, err := s.FindUserResourceMappings(ctx, influxdb.UserResourceMappingFilter{
+			ResourceType: influxdb.BucketsResourceType,
+			ResourceID:   data.Buckets[1].ID,
+		})
+		if nurms != 0 {
+			t.Errorf("expected bucket2, to be dangling, got: %+v", urms)
 		}
 		// Now delete user1 -> bucket1.
-		// bucket1 should be deleted (nobody points to it).
+		// Still expect bucket1 to exist (nobody points to it).
 		if err := s.DeleteUserResourceMapping(ctx, data.Buckets[0].ID, data.Users[0].ID); err != nil {
 			t.Fatal(err)
 		}
 		bs, nbs, err = s.FindBuckets(ctx, influxdb.BucketFilter{ID: &data.Buckets[0].ID})
 		if err != nil {
 			t.Fatal(err)
 		}
-		if nbs > 0 {
-			t.Errorf("expected no buckets, got: %v", bs)
-			urms, _, err := s.FindUserResourceMappings(ctx, influxdb.UserResourceMappingFilter{
-				ResourceType: influxdb.BucketsResourceType,
-				ResourceID:   data.Buckets[1].ID,
-			})
-			if err != nil {
-				t.Fatal(err)
-			}
-			t.Logf("bucket1 should be dangling at this point. These are its urms: %+v", urms)
+		if nbs != 1 {
+			t.Errorf("expected 1 buckets, got: %v", bs)
+		}
+		urms, nurms, err = s.FindUserResourceMappings(ctx, influxdb.UserResourceMappingFilter{
+			ResourceType: influxdb.BucketsResourceType,
+			ResourceID:   data.Buckets[0].ID,
+		})
+		if nurms != 0 {
+			t.Errorf("expected bucket1, to be dangling, got: %+v", urms)
 		}
 	})
 
-	// NOTE(affo)(*kv.Service): it deletes urms, but not dangling resources.
-	t.Run("deleting a user deletes every urm and so for dangling resources", func(t *testing.T) {
+	t.Run("deleting a user deletes every related urm and nothing else", func(t *testing.T) {
```

Comment:
decided not  to  remove dangling resources, that seems too much.

My thought  is that we do not want to automatically propagate this kind of delete because a resource will always have an owner org.